lib/rules/no-broken-super-chain.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @fileOverview Prevent the absence of this._super() in init() calls or the use of this prior to this._super()
+ * @author Quinn C Hoyer
+ */
+
+const MESSAGES = {
+  noSuper: '\'this._super(...arguments)\' must be called in init in this Ember modules',
+  noThisBeforeSuper: 'Must call \'this._super(...arguments)\' before accessing `this`',
+  tooManySupers: 'Only call `this._super(...arguments)` once per init()'
+};
+
+// TODO: Make this configurable
+const EMBER_MODULES_WITH_SUPER_CHAIN = {
+  Component: true,
+  Mixin: true,
+  Route: true,
+  Controller: true,
+  View: true
+};
+
+let initOverride = null;
```

Comment:
Could we initialize this in the `create()` function? Just to ensure it has a more specific scoping.